File: poppy.js
const letters = 'Poppy'.split('');
const N = 3e6;

for (let i = 0; i < N; i++) {
  for (let j = 0; j <= i; j++) {
    process.stdout.write(letters[j % 5]);
  }
  process.stdout.write('\n');
}

File: poppy.scala
object Main extends App {
	def poppy = (i: Int) => {
		val p = "Poppy".split("").toArray
		List.tabulate(i)(n => n)
			.foreach(el => {
				print(p(el % 5))
			})
			println
	}

	List.tabulate(3000000)(n => n + 1)
		.foreach(poppy)
}